File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/validation/AbstractValidationRule.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.validation;

import java.io.Serializable;
import java.security.InvalidParameterException;

/**
 * Abstract class which contains functionality to handle validation.
 */
public abstract class AbstractValidationRule implements Serializable, ValidationRule  {

	private static final long serialVersionUID = -1068723487645115780L;

	// Validation message that must be translated
	private String messageId;

	// Validationtype
	private ValidationType validationType;

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public AbstractValidationRule(String messageId, ValidationType type) {

		if (messageId == null) {
			throw new InvalidParameterException("Error initialising ValidationRule, messageId may not be null");
		}
		if (type == null) {
			throw new InvalidParameterException("Error initialising ValidationRule, type may not be null");
		}
		this.messageId = messageId;
		this.validationType = type;
	}


	public String getMessageId() {
		return messageId;
	}

	public ValidationType getType() {
		return validationType;
	}

}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/validation/AbstractValidationRule.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.validation

import com.onlinepayments.sdk.client.android.model.PaymentRequest
import java.io.Serializable

/**
 * Abstract class which contains functionality to handle validation.
 */
abstract class AbstractValidationRule internal constructor(
    @Suppress("Unused")
    val messageId: String,
    val type: ValidationType
) : Serializable, ValidationRule {

    protected fun getUnmaskedValue(paymentRequest: PaymentRequest, fieldId: String): String? {
        var text = paymentRequest.getValue(fieldId) ?: return null

        return paymentRequest.getUnmaskedValue(fieldId, text)
    }

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = -1068723487645115780L
    }
}
